thingif/src/main/java/com/kii/thingif/clause/query/RangeClauseInQuery.java

```diff
@@ -0,0 +1,118 @@
+package com.kii.thingif.clause.query;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.kii.thingif.clause.base.BaseRange;
+
+import org.json.JSONObject;
+
+public class RangeClauseInQuery implements QueryClause, BaseRange {
+    private @NonNull String field;
+    private @Nullable Long upperLimit;
+    private @Nullable Long lowerLimit;
+    private @Nullable Boolean upperIncluded;
+    private @Nullable Boolean lowerIncluded;
+
+    private RangeClauseInQuery(
+            @NonNull String field,
+            @Nullable Long upperLimit,
+            @Nullable Boolean upperIncluded,
+            @Nullable Long lowerLimit,
+            @Nullable Boolean lowerIncluded) {
+        this.field = field;
+        this.upperIncluded = upperIncluded;
+        this.upperLimit = upperLimit;
+        this.lowerIncluded = lowerIncluded;
+        this.lowerLimit = lowerLimit;
+    }
+    public static RangeClauseInQuery range(
+            @NonNull String field,
+            @NonNull Number upperLimit,
+            @NonNull Boolean upperIncluded,
+            @NonNull Number lowerLimit,
+            @NonNull Boolean lowerIncluded) {
+        return new RangeClauseInQuery(
+                field,
+                upperLimit.longValue(),
+                upperIncluded,
+                lowerLimit.longValue(),
+                lowerIncluded);
+    }
+    public static RangeClauseInQuery greaterThan(
+            @NonNull String field,
+            @NonNull Number lowerLimit) {
+        return new RangeClauseInQuery(
+                field,
+                null,
+                null,
+                lowerLimit.longValue(),
+                Boolean.FALSE);
+    }
+    public static RangeClauseInQuery greaterThanEquals(
```

Comment:
May be better to follow here: https://docs.oracle.com/javase/tutorial/java/nutsandbolts/opsummary.html